### PR TITLE
Merge Julia to Dev

### PR DESCRIPTION
**Pull request by pioupiou-devs**

### CODE REVIEW

File: src/client/requirements.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+ * Send json data to target url
+ * @param {String} url
+ * @param {String} data
+ * @param {String} method
+ * @returns {String} Response JSON
+ */
+async function sendRequest(url, data={}, method) {
+    const response = await fetch(url, {
+        method: method,
+        mode: 'cors',
```
Comment (pioupiou-devs) [inline src/client/requirements.js]: CORS problem i think, see with @andrbtn

File: src/client/requirements.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    });
+    return response.json();
+}
+
+/**
+ * Check form inputs and sends a JSON with all data to server
+ * @returns {boolean} True if all inputs are valid, else False
+ */
+function setRequirements() {
+    let inputNbPlayers = +document.getElementById("nbPlayers").value;
```
Comment (pioupiou-devs) [inline src/client/requirements.js:62]: strange space between = and +

File: src/client/requirements.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    let inputNames = document.getElementById("playersNames").value.split(',');
+    inputNames = inputNames.map(name => {
+        return name.trim();
+    });
+    let inputNbRounds = +document.getElementById("nbRounds").value;
```
Comment (pioupiou-devs) [inline src/client/requirements.js:67]: strange space

File: src/client/requirements.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    let inputNbPins = +document.getElementById("nbPins").value;
```
Comment (pioupiou-devs) [inline src/client/requirements.js:68]: strange space

File: src/client/requirements.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    } else if (inputNbPlayers > 1 && containDuplicateNames(inputNames)) {
+        showErrorText("Some players have the same name.")
+        return false;
+    } else {
+        let redirectURL = location.href.substring(0, location.href.lastIndexOf("/")+1) + "index.html";
+        let data = JSON.stringify({nbKeel:inputNbPins, nbFrames:inputNbRounds, players:inputNames, "redirect":redirectURL});
+        console.log(data);
+
+        // Send json to server
+        sendRequest('http://localhost:3000/api/grid', 'PUT')
```
Comment (pioupiou-devs) [inline src/client/requirements.js]: Can you add a global variable for the server url please ?

File: src/client/requirements.js (hunk 2 above)
Comment (Pyrignos) [reply src/client/requirements.js:62]: '+' is there to force conversion of value to int

File: src/client/requirements.js (hunk 3 above)
Comment (Pyrignos) [reply src/client/requirements.js:67]: (same as above)

'+' is there to force conversion of value to int

File: src/client/requirements.js (hunk 4 above)
Comment (Pyrignos) [reply src/client/requirements.js:68]: (same as above)

'+' is there to force conversion of value to int

File: src/client/requirements.js (hunk 5 above)
Comment (Pyrignos) [reply src/client/requirements.js]: Done

File: src/client/requirements.js (hunk 1 above)
Comment (Pyrignos) [reply src/client/requirements.js]: changed mode to "no-cors"

**Pull request merged**